src/core/queue/DatabaseQueue.ts

```diff
@@ -0,0 +1,34 @@
+/* eslint-disable no-shadow */
+import BaseQueue from './BaseQueue';
+
+export enum DatabaseOperationLabel {
+  CoreChatSystem_SaveFollowingMessage = 'core_chat_system_save_following_message',
+  CoreChatSystem_GeneralSystemMessage = 'core_chat_system_general_system_message',
+  CoreChatSystem_SaveChat = 'core_chat_system_save_chat',
+  CoreChatSystem_SaveUserMember = 'core_chat_system_save_user_member',
+
+  FetchChannelDetail_SaveAllChat = 'channel_detail_save_all_chat',
+  FetchChannelDetail_RefreshChannelList = 'channel_detail_refresh_channel_list',
+  FetchChannelDetail_SaveChannelList = 'channel_detail_save_channel_list',
+  FetchChannelDetail_SaveUserMember = 'channel_detail_save_user_member',
+
+  ChatScreen_GetChat = 'chat_screen_get_chat',
+  ChatScreen_GetSelfAnonUserInfo = 'chat_screen_get_self_anon_user_info',
+  ChatScreen_SendChat = 'chat_screen_send_chat',
+  ChatScreen_UpdateChannelDescription = 'chat_screen_update_channel_description',
+  ChatScreen_UpdateChatSentStatus = 'chat_screen_update_chat_sent_status'
+}
+
+class DatabaseQueue extends BaseQueue {
+  static dbInstance: DatabaseQueue;
+
+  static getInstance(): DatabaseQueue {
+    if (!this.dbInstance) {
+      return new DatabaseQueue();
+    }
+
+    return this.dbInstance;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The singleton pattern implementation in `DatabaseQueue` class seems incorrect. The created instance of `DatabaseQueue` is not being assigned to `dbInstance`, so every time `getInstance()` is called, a new instance will be returned instead of the same instance.

Here's how you can fix it:

```diff
  static getInstance(): DatabaseQueue {
    if (!this.dbInstance) {
-       return new DatabaseQueue();
+       this.dbInstance = new DatabaseQueue();
    }

    return this.dbInstance;
  }
```
This way, the first call to `getInstance()` will create a new instance and assign it to `dbInstance`. Subsequent calls will return the existing instance, ensuring that only one instance of `DatabaseQueue` exists throughout the application.


<!-- This is an auto-generated comment by OSS CodeRabbit -->